torch/distributed/_functional_collectives_impl.py

```diff
@@ -304,3 +304,32 @@ def _reduce_scatter_tensor_coalesced_fallback(output_tensors, input_tensors, op,
         work = c10d.reduce_scatter_tensor(out_tensor, shard, op=op, group=group, async_op=async_op)
         work_list.append(work)
     return work_list
+
+
+def _all_to_all_single(
+    input: torch.Tensor,
+    output_split_sizes: Optional[torch.Tensor],
+    input_split_sizes: Optional[torch.Tensor],
+    tag: str,
+    ranks: List[int],
+    group_size: int,
+):
+    group = c10d._find_or_create_pg_by_ranks_and_tag(tag, ranks, group_size)
+
+    if isinstance(input_split_sizes, torch.Tensor):
+        input_split_sizes = input_split_sizes.tolist()
```

Comment:
Hmm. Is there any particular reason you call tolist here, as opposed to just taking a list of ints as arguments (you call tolist outside of the collective). Is it because the list could be very long?

Reply:
I believe Dynamo will hardcode the `List[int]` input arguments as constants instead of treating them as dynamic inputs like tensors, and it's why I make `input_split_sizes` and `output_split_sizes` tensors instead

Reply:
If it's `SymInt[]` it wouldn't be hardcoded. If you're actually extracting out the tensor values to ints at runtime it would be better to use this.

Reply:
discussed with Ed offline - will use tensor as input type instead, because list-type input to `output_split_sizes` doesn’t have a special data structure like KJT that we can insert conversion logic for converting ints to unbacked symints.